app/commands/serial_sequence_command.rb

```diff
@@ -31,15 +31,20 @@ def validate_all_commands
   end
 
   def validate_command(command)
+    pos = data.index { |c| c.uuid == command.uuid }
```

Comment:
There was a bug in this code - the idx was being used to report the position in the data array of the error - however, the idx is NOT this - it is the index in the collection of errors - used for getting the message.
So now the idx is calculated by searching for the uuid in the data